rawproduce.go

```diff
@@ -0,0 +1,103 @@
+package kafka
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net"
+
+	"github.com/segmentio/kafka-go/protocol"
+	produceAPI "github.com/segmentio/kafka-go/protocol/produce"
+	"github.com/segmentio/kafka-go/protocol/rawproduce"
+)
+
+// RawProduceRequest represents a request sent to a kafka broker to produce records
+// to a topic partition. The request contains a pre-encoded/raw record set.
+type RawProduceRequest struct {
+	// Address of the kafka broker to send the request to.
+	Addr net.Addr
+
+	// The topic to produce the records to.
+	Topic string
+
+	// The partition to produce the records to.
+	Partition int
```

Comment:
I know our existing `ProduceRequest` only supports producing to one partition at a time, but should we take this opportunity to add support for producing to multiple partitions in one request with `RawProduceRequest`? This could result in a large performance improvement and reduction in load on Kafka clusters for clients that can accommodate for it

Reply:
If that is too much unrelated work for this feature we could always add some kind of `MultiRawProduceRequest` later on if needed